x/evm/keeper/smart_contract_deployment.go

```diff
@@ -57,15 +58,15 @@ func (k Keeper) DeleteSmartContractDeploymentByContractID(ctx sdk.Context, smart
 }
 
 func (k Keeper) SetSmartContractDeploymentStatusByContractID(ctx sdk.Context, smartContractID uint64, chainReferenceID string, status types.SmartContractDeployment_Status) error {
-	logger := k.Logger(ctx).WithFields("smart-contract-id", smartContractID, "chain-reference-id", chainReferenceID, "new-smart-contract-status", status)
+	logger := k.Logger(ctx).With("smart-contract-id", smartContractID, "chain-reference-id", chainReferenceID, "new-smart-contract-status", status)
```

Comment:
Make sure to use `WithFields` for setting logging fields. It returns a different type than the default `With()` and we want that.

Reply:
@byte-bandit As we have coverted to cosmossdk.io/log ...there are no WithFields function in logger...should i just use the with??

Reply:
Use the `liblog` package for this, it wraps the underlaying logger.